### PR TITLE
Ch12 test win fix

### PR DESCRIPTION
**Pull request by othelarian**
Fix following issues:

* On Windows, if git is configured to automatically do the "\r\n" conversion, `foo.txt` come with an extra character, invalidating `countCharacters` test
* On Windows, when using standard console, the `ENOENT` error only use absolute path instead of relative path
* Behind an http-only proxy, the https://reqres.in isn't reachable

NOTE : for the "behind http-only proxy" issue, maybe a note or a warning in the book will be relevant to prevent user wasting time finding why proxy doesn't work.

### CODE REVIEW

File: exercises/chapter12/test/Main.purs [hunk 1]
```diff
@@ -28,6 +30,15 @@ inDir = Path.concat [ "test", "data" ]
 outDir :: FilePath
 outDir = Path.concat [ "test", "data-out" ]
 
+-- If you're behind an http-only proxy, the call to https://reqres.in won't work.
+-- A work around is to install http-server (npm i -g http-server) and run it
+-- locally in the "test/data" directory (http-server -p 42524).
+reqUrl :: String
+reqUrl =
+  "https://reqres.in/api/users/1"
```
Comment (milesfrain) [inline exercises/chapter12/test/Main.purs:38]: Will changing this to the non-https version work?
```
http://reqres.in/api/users/1
```

File: exercises/chapter12/test/Main.purs [hunk 2]
```diff
@@ -67,16 +78,21 @@ main =
     suite "countCharacters" do
       test "exists" do
         chars <- countCharacters $ Path.concat [ inDir, "foo.txt" ]
```
Comment (milesfrain) [inline exercises/chapter12/test/Main.purs:80]: What about just changing `foo.txt` so there are no line breaks?

**Comment by milesfrain** [comment]
Thanks. Looks good overall, although I'm wondering if there are simpler fixes for the `https` and `\r\n` issues.
Also never heard of "http-only proxies" before. What are those used for?

File: exercises/chapter12/test/Main.purs (hunk 2 above)
Comment (othelarian) [reply exercises/chapter12/test/Main.purs:80]: Yes, it will do the trick too

File: exercises/chapter12/test/Main.purs (hunk 1 above)
Comment (othelarian) [reply exercises/chapter12/test/Main.purs:38]: Maybe, but as every api is or will be in https soon, I think this isn't a good idea to lower the security instead of adding less than 5 lines of code. I also see an interested feature as it allows offline usability of the test. Where I live there is a lot of area without connectivity, and being able to use a localhost is, imo, a real benefit.

Maybe I can change the comments above and push for the offline way, and some edge case about http-only proxy?

**Comment by othelarian** [comment]
As english isn't my fist language, I found someone who can explain the [issue better than me](https://janmolak.com/node-js-axios-behind-corporate-proxies-8b17a6f31f9d?gi=485ea3cefe7f)

In short, legacy proxy with no one to update it. It sadly happens quite often in France as the momentum in old companies tends to stuck them with old software.

For the `\r\n` I think the solution you propose is better, I'll make the change.

For the localhost idea, I think offline capability can be useful, not only for the edge case of http-only proxy, but also for offline people. What do you think?

**Comment by milesfrain** [comment]
> For the localhost idea, I think offline capability can be useful

That's true.
I've also had outages with these API examples. https://jsonplaceholder.typicode.com/ was causing lots of problems. http://reqres.in/api has been reliable so far.
So a comment describing the `localhost` workaround with `http-server` seems good.
I don't think offline should be the default though, since that would require some extra setup steps.

File: exercises/chapter12/test/Main.purs [hunk 3]
```diff
@@ -28,6 +30,15 @@ inDir = Path.concat [ "test", "data" ]
 outDir :: FilePath
 outDir = Path.concat [ "test", "data-out" ]
 
+-- If you're behind an http-only proxy, the call to https://reqres.in won't work.
```
Comment (milesfrain) [inline exercises/chapter12/test/Main.purs:33]: Thinking something like this for instructions:
> -- If you'd like to run this test offline, you can create this API endpoint locally by installing http-server (npm i -g http-server) and runing it in the "test/data" directory (http-server -p 42524).
-- If you can't reach https://reqres.in due to http-only proxy issues, you may either use the above `http-server` solution, or simply remove the `s` from `https` in the url:
--  reqUrl = http://reqres.in/api/users/1

**Comment by othelarian** [comment]
@milesfrain you can review, I added what you noticed, changed for a new text file without linebreak, and updated the comments to avoid talking about edge cases and more about offline. I also add a comment about the use of http instead of https for reqres.in ;-)

**Comment by milesfrain** [comment]
Thanks. This looks great.
I added a few minor adjustments.
